### PR TITLE
Change SGD with momentum to include momentum correction by default

### LINKED ISSUE #15533: The problems about SGD with momentum when learning rate changes
Hi, there.
Currently, SGD(Stochastic Gradient Descent) in MXNet is applied by:
```
rescaled_grad = lr * (rescale_grad * clip(grad, clip_gradient) + wd * weight)
state = momentum * state + rescaled_grad
weight = weight - state
```
I found there are two problems in SGD.
1. Loss of accuracy on float-point number
For the SGD with momentum, the variable `state` stores the gradients multiplied by learning rate. However, learning rate is usually a small value, such as 1e-3, enabling the `state` becomes smaller than the gradient. It may loss the accuracy.

2. The case when learning rate changes.
When learning rate changes, the variable `state` stores the gradients multiplied by old learning rate. It is wrong.

Solution:
We should update the implement of SGD with momentum, but we should consider the compatibility with old optimizer states.
```
rescaled_grad = rescale_grad * clip(grad, clip_gradient) + wd * weight
state = momentum * state + rescaled_grad
weight = weight - lr * state
```

### PR DESCRIPTION
**Pull request by ptrendx**
## Description ##
Fixes #15533

Notable change needed to happen to LARS optimizer, since it actually requires the lack of momentum correction, so I had to emulate it in the unaggregated case. I also moved and renamed contrib/preloaded_*_sgd -> lars_*_sgd operators (moved out of contrib because non-contrib functionality depends on them).

@wkcn @zhreshold @shengac FYI

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- [x] Code is well-documented: 
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: python/mxnet/optimizer/lars.py [hunk 1]
```diff
@@ -252,6 +253,13 @@ def fused_step(self, indices, weights, grads, states):
                 wd = wds[i]
                 lr = lrs[i]
                 lr *= self._get_lars(index, weight, grad, wd)
+                # normal SGD picks up momentum correction by default,
+                # so need to modify the momentum to undo that.
+                # The correction term is previous_lr / current_lr.
+                kwargs['momentum'] = (self.momentum * (self.last_lr / lr)) \
```
Comment (wkcn) [inline python/mxnet/optimizer/lars.py:259]: Thank you for the fix!

It will reduce the times of getting an item of `self.last_lr`.
```python
kwargs['momentum'] = (self.momentum * (self.last_lr.get(index, lr) / lr)) \
                                     if lr != 0 else \
                                     self.momentum
```

Comment (ptrendx) [reply python/mxnet/optimizer/lars.py]: Thanks, done

**Comment by zhreshold** [comment]
@szhengac Can you provide feedback on this?

**Comment by szhengac** [comment]
Overall, it looks good to me. It overlaps partially with #18285.
